lib/spack/spack/cmd/diy.py

```diff
@@ -52,9 +52,14 @@ def setup_parser(subparser):
     subparser.add_argument(
         'spec', nargs=argparse.REMAINDER,
         help="specs to use for install.  Must contain package AND version.")
-    subparser.add_argument(
-        '--dirty', action='store_true', dest='dirty',
-        help="Install a package *without* cleaning the environment.")
+
+    cd_group = subparser.add_mutually_exclusive_group()
+    cd_group.add_argument(
+        '--clean', action='store_false', dest='dirty', default=None,
+        help="Clean environment before installing package.")
+    cd_group.add_argument(
+        '--dirty', action='store_true', dest='dirty', default=None,
+        help="Do NOT clean environment before installing.")
 
```

Comment:
These command args are common to many commands.  Can they be factored out in a subroutine used ball all cmds that need them?